qemu/tests/hotplug_virtio_mem.py

```diff
@@ -0,0 +1,55 @@
+import time
+
+from virttest import error_context
+
+from virttest.utils_misc import normalize_data_size
+from provider import virtio_mem_utils
+from virttest.utils_test.qemu import MemoryHotplugTest
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Virtio-mem hotplug test
+    1) Boot guest
+    2) Hotplug virtio-mem device
+    3) Check virtio-mem device
+    4) Resize virtio-mem device twice
+    5) Check virtio-mem device
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    timeout = int(params.get("login_timeout", 240))
+    threshold = params.get_numeric("threshold", target_type=float)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    virtio_mem_model = 'virtio-mem-pci'
+
+    target_mem = params.get("target_mems")
+    hotplug_test = MemoryHotplugTest(test, params, env)
+    hotplug_test.hotplug_memory(vm, target_mem)
+
+    if '-mmio:' in params.get("machine_type"):
+        virtio_mem_model = 'virtio-mem-device'
+    for i, vmem_dev in enumerate(vm.devices.get_by_params({'driver': virtio_mem_model})):
+        device_id = vmem_dev.get_qid()
+        requested_size_vmem = params.get("requested-size_test_vmem%d" % i)
+        node_id = int(vmem_dev.params.get("node"))
+        for requested_size in requested_size_vmem.split():
+            req_size_normalized = int(float(normalize_data_size(requested_size,
+                                                                'B')))
+            vm.monitor.qom_set(device_id, "requested-size",
+                               req_size_normalized)
+            time.sleep(30)
+            virtio_mem_utils.check_memory_devices(device_id, requested_size,
+                                                  threshold, vm, test)
+            virtio_mem_utils.check_numa_plugged_mem(node_id, requested_size,
```

Comment:
I think it's better to use newly added memory after hotplug_memory
like utils_test.VMStress or 'numactl -m %s dd if=/dev/urandom of=/tmp/numa_test/test '

On the other hand we are not going to test unplug?

Reply:
@zhenyzha 
> I think it's better to use newly added memory after hotplug_memory like utils_test.VMStress or 'numactl -m %s dd if=/dev/urandom of=/tmp/numa_test/test '
> 

Mmm interesting point, the polarion case only reflects to test the hotplug but not using the new memory, also then I will need to modify the manual case as well... do you think it's better to include that in this thread, or in a future one?

> On the other hand we are not going to test unplug?

As confirmed with the developer and reflected in the [documentation](https://docs.google.com/document/d/1LG9Vqm6Q3TKX5X7__RpsnN9K13innr_9H0XVLUELT_M/edit#heading=h.c9xunb4skuja), the virtio-mem unplug is not supported

Reply:
Keeping it consistent with the polarion case is ok for me, 
I think we can add it to the future plan.